src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -135,15 +139,33 @@ public Test BuildFrom(IMethodInfo method, Test parentSuite)
             if (needCombinatorial)
                 builders.Add(new CombinatorialAttribute());
 
+            var exceptions = new List<Exception>();
+
             foreach (var attr in builders)
             {
-                foreach (var test in attr.BuildFrom(method, parentSuite))
-                    tests.Add(test);
+                try
+                {
+                    foreach (var test in attr.BuildFrom(method, parentSuite))
+                        tests.Add(test);
+                }
+                catch (Exception ex)
+                {
+                    exceptions.Add(ex);
+                }
```

Comment:
Handling exceptions for this purpose has precedent in NUnit. The other option is to return more info from BuildFrom, or somehow give BuildFrom an instance of the test suite so it can be modified.